OCCommunicationLib/OCCommunicationLib/OCCommunication.m

```diff
@@ -555,38 +555,42 @@ - (void) setDownloadTaskDidGetBodyDataBlock: (void(^)(NSURLSession *session, NSU
+- (NSURLSessionUploadTask *) uploadFile:(NSString *) localPath toDestiny:(NSString *) remotePath onCommunication:(OCCommunication *)sharedOCCommunication withProgress:(NSProgress * __autoreleasing *) progressValue successRequest:(void(^)(NSURLResponse *response, NSString *redirectedServer)) successRequest failureRequest:(void(^)(NSURLResponse *response, NSString *redirectedServer, NSError *error)) failureRequest failureBeforeRequest:(void(^)(NSError *error)) failureBeforeRequest {
     
-    remotePath = [remotePath encodeString:NSUTF8StringEncoding];
+    OCWebDAVClient *request = [[OCWebDAVClient alloc] initWithBaseURL:[NSURL URLWithString:@""]];
+    request = [self getRequestWithCredentials:request];
+    request.securityPolicy = self.securityPolicy;
     
-    OCUploadOperation *operation = [OCUploadOperation new];
+    remotePath = [remotePath encodeString:NSUTF8StringEncoding];
     
-    [operation createOperationWith:localPath toDestiny:remotePath onCommunication:sharedOCCommunication progressUpload:^(NSUInteger bytesWrote, long long totalBytesWrote, long long totalBytesExpectedToWrote) {
-        progressUpload(bytesWrote, totalBytesWrote, totalBytesExpectedToWrote);
-    } successRequest:^(NSHTTPURLResponse *response, NSString *redirectedServer) {
-        successRequest(response, redirectedServer);
-    } failureRequest:^(NSHTTPURLResponse *response, NSData *responseData, NSString *redirectedServer, NSError *error) {
-        
-        OCXMLServerErrorsParser *serverErrorParser = [OCXMLServerErrorsParser new];
-        
-        [serverErrorParser startToParseWithData:responseData withCompleteBlock:^(NSError *err) {
-            
-            if (err) {
-                failureRequest(response, redirectedServer, err);
-            }else{
-                failureRequest(response, redirectedServer, error);
-            }
-            
-        }];
-        
-        
-    } failureBeforeRequest:^(NSError *error) {
-        failureBeforeRequest(error);
-    } shouldExecuteAsBackgroundTaskWithExpirationHandler:^{
-        handler();
-    }];
+    NSURLSessionUploadTask *uploadTask = [request putLocalPath:localPath atRemotePath:remotePath onCommunication:sharedOCCommunication withProgress:progressValue
+                                                                  success:^(NSURLResponse *response, id responseObjec){
+                                                                      [UtilsFramework addCookiesToStorageFromResponse:(NSURLResponse *) response andPath:[NSURL URLWithString:remotePath]];
+                                                                      //TODO: The second parameter is the redirected server
+                                                                      successRequest(response, @"");
+                                                                  } failure:^(NSURLResponse *response, id responseObject, NSError *error) {
```

Comment:
You could use responseObject as NSData in this line so that you don't need to use the variable responseData after. Same issue in method : 

(NSURLSessionUploadTask *) uploadFileSession:(NSString *) localPath toDestiny:(NSString *) remotePath onCommunication:(OCCommunication *)sharedOCCommunication withProgress:(NSProgress * __autoreleasing *) progressValue successRequest:(void(^)(NSURLResponse *response, NSString *redirectedServer)) successRequest failureRequest:(void(^)(NSURLResponse *response, NSString *redirectedServer, NSError *error)) failureRequest failureBeforeRequest:(void(^)(NSError *error)) failureBeforeRequest